internal/controller/controllers/agentserviceconfig_controller_test.go

```diff
@@ -104,14 +106,38 @@ var _ = Describe("agentserviceconfig_controller reconcile", func() {
 
 	BeforeEach(func() {
 		asc = newASCDefault()
-		ascr = newTestReconciler(asc, ingressCM, route, imageRoute)
 	})
 
 	It("reconcile should succeed", func() {
+		agentinstalladmissionDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "agentinstalladmission",
+				Namespace: "assisted-installer",
+			},
+		}
+		imageServiceDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "assisted-image-service",
+				Namespace: "assisted-installer",
+			},
+		}
+		assistedServiceDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "assisted-service",
+				Namespace: "assisted-installer",
+			},
+		}
+		ascr = newTestReconciler(asc, ingressCM, route, imageRoute, agentinstalladmissionDeployment, imageServiceDeployment, assistedServiceDeployment)
 		result, err := ascr.Reconcile(ctx, newAgentServiceConfigRequest(asc))
 		Expect(err).To(Succeed())
 		Expect(result).To(Equal(ctrl.Result{}))
 	})
+	It("reconcile should fail", func() {
```

Comment:
What is this testing?
"reconcile should fail ..." under what conditions?

Reply:
It should fail when the deployments are not created/available. 

In `"reconcile should succeed"` here https://github.com/openshift/assisted-service/pull/3031/files#diff-e2012bf81c2143a9e5dcb82567f9c141cf44e6361b26678daf19eac711097910R111, `newTestReconciler` creates the deployments for the test case purpose but are not passed to `newTestReconciler` in  `reconcile should fail` case